llama_index/vector_stores/google/generativeai/base.py

```diff
@@ -0,0 +1,231 @@
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        _logger.debug(f"\n\nGoogleVectorStore.query(query={query})")
+
+        client = cast(genai.RetrieverServiceClient, self.client)
+
+        relevant_chunks: List[genai.RelevantChunk] = []
+        if query.doc_ids is None:
+            relevant_chunks = genaix.query_corpus(
+                corpus_id=self.corpus_id,
+                query=query.query_str or "what?",
+                filter=_convertFilter(query.filters),
+                k=query.similarity_top_k,
+                client=client,
+            )
+        else:
+            for doc_id in query.doc_ids:
+                relevant_chunks.extend(
+                    genaix.query_document(
+                        corpus_id=self.corpus_id,
+                        document_id=doc_id,
+                        query=query.query_str or "what?",
+                        filter=_convertFilter(query.filters),
+                        k=query.similarity_top_k,
+                        client=client,
+                    )
+                )
+
+        return VectorStoreQueryResult(
+            nodes=[
+                TextNode(
+                    text=chunk.chunk.data.string_value,
+                    id_=_extract_chunk_id(chunk.chunk.name),
+                )
+                for chunk in relevant_chunks
+            ],
+            ids=[_extract_chunk_id(chunk.chunk.name) for chunk in relevant_chunks],
+            similarities=[chunk.chunk_relevance_score for chunk in relevant_chunks],
+        )
+
+
+def _extract_chunk_id(entity_name: str) -> str:
+    try:
+        import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+    except ImportError:
+        raise ImportError(_import_err_msg)
+
+    id = genaix.EntityName.from_str(entity_name).chunk_id
+    assert id is not None
+    return id
+
+
+class _NodeGroup(BaseModel):
+    """Every node in nodes have the same source node."""
+
+    source_node: RelatedNodeInfo
+    nodes: List[BaseNode]
+
+
+def _groupNodesBySource(nodes: Sequence[BaseNode]) -> List[_NodeGroup]:
+    """Returns a list of lists of nodes where each list has all the nodes
+    from the same document.
+    """
+    groups: Dict[str, _NodeGroup] = {}
+    for node in nodes:
+        source_node: RelatedNodeInfo
+        if isinstance(node.source_node, RelatedNodeInfo):
+            source_node = node.source_node
+        else:
+            source_node = RelatedNodeInfo(node_id=_default_doc_id)
+
+        if source_node.node_id not in groups:
+            groups[source_node.node_id] = _NodeGroup(source_node=source_node, nodes=[])
+
+        groups[source_node.node_id].nodes.append(node)
+
+    return list(groups.values())
+
+
+def _convertFilter(fs: Optional[MetadataFilters]) -> Dict[str, Any]:
```

Comment:
should we add a comment on the filter on what's supported? or a link to documentation might be better